yang/IETF-04/ietf-vpn-types.yang

```diff
@@ -0,0 +1,346 @@
+
+     Redistribution and use in source and binary forms, with or
+     without modification, is permitted pursuant to, and subject to
+     the license terms contained in, the Simplified BSD License set
+     forth in Section 4.c of the IETF Trust's Legal Provisions
+     Relating to IETF Documents
+     (https://trustee.ietf.org/license-info).
+
+     This version of this YANG module is part of RFC XXXX
+     (https://www.rfc-editor.org/info/rfcXXXX); see the RFC itself
+     for full legal notices.";
+
+  revision 2020-06-03 {
+    description
+      "Initial revision.";
+    reference
+      "RFC XXXX: A VPN common types YANG Model";
+      // RFC Ed.: replace XXXX with actual RFC number and remove
+     // this note
+  }
+
+/* Feature */
+
+ feature cloud-access {
+  description
+  "Allows the VPN to connect to a CSP.";
+ }
+
+/* Typedef */
+
+typedef operational-type {
+    type enumeration {
+      enum up {
+        value 0;
+        description
+          "Operational status UP/Enabled.";
+      }
+      enum down {
+        value 1;
+        description
+          "Operational status DOWN/Disabled.";
+      }
+      enum unknown {
+        value 2;
+        description
+          "Operational status UNKNOWN.";
+      }
+    }
+    description
+      "This attribute is used to determine the
+       status of a particular element.";
+  }
+
+	/* Identities */
+
+  identity encapsulation-type {
+      description
+      "Base identity for Protocol Type.";
+  }
+
+  identity priority-tagged {
+    base encapsulation-type;
+    description
+      "Identity for the priority-tagged interface.";
+  }
+
+  identity dot1q {
+  	base encapsulation-type;
+    description
+      "This identity indicates the support of
+       the 'dot1q' encapsulation.";
+  }
+
+  identity qinq {
+  	base encapsulation-type;
+    description
+      "This identity indicates the support of
+       the 'qinq' encapsulation.";
+  }
+
+  identity qinany {
+  	base encapsulation-type;
+    description
+      "This identity indicates the support of
+       the 'qinany' encapsulation.";
+  }
+
+  identity vxlan {
+  	base encapsulation-type;
+    description
+      "This identity indicates the support of
+       the 'vxlan' encapsulation.";
+  }
+
+  identity tag-type {
+    description
+      "Base identity from which all tag types are derived.";
+  }
+
+  identity c-vlan {
+    base tag-type;
+    description
+      "A CVLAN tag, normally using the 0x8100 Ethertype.";
+  }
+
+  identity s-vlan {
+    base tag-type;
+    description
+      "An SVLAN tag.";
+  }
+
+  identity c-s-vlan {
+    base tag-type;
+    description
+      "Using both a CVLAN tag and an SVLAN tag.";
+  }
+
+identity protocol-type {
+      description
+      "Base identity for Protocol Type.";
+  }
+
+  identity GRE {
+    base protocol-type ;
+    description
+      "Transport based on GRE.";
+  }
+
+  identity LDP {
+    base protocol-type ;
+    description
+      "Transport based on LDP.";
```

Comment:
We need to provide a description of what is meant by transport on LDP. Please add a reference clause.